stream.go

```diff
@@ -0,0 +1,769 @@
+package http2
+
+import (
+	"container/list"
+	"fmt"
+)
+
+const (
+	defaultWeight    = 15
+	defaulMaxDepTree = 100
+	maxStreamWeight  = 255
+)
+
+var defaultPriority = PriorityParam{
+	Weight:    defaultWeight,
+	StreamDep: 0,
+}
+
+type streamDepState int
+
+const (
+	depStateIdle     streamDepState = iota // init state without data
+	depStateReady                          // ready to send but blocked by dependency tree
+	depStateTop                            // can send data
+	depStatFlowDefer                       // blocked by stream flow control
+)
+
+var depStateName = [...]string{
+	depStateIdle:     "Idle",
+	depStateReady:    "Ready",
+	depStateTop:      "Top",
+	depStatFlowDefer: "FlowDefer",
+}
+
+func (ds streamDepState) String() string {
+	return depStateName[ds]
```

Comment:
maybe a range check like this:
```go
	if 0 <= ds && ds < streamDepState(len(depStateName)) {
		return depStateName[ds]
	}
	return fmt.Sprintf("http2: unknown streamDepState %d", ds)
```
here in order to handle it like this: http://play.golang.org/p/EIhDuHDq9n instead of panic might ease debugging?